### DISCUSSION TITLE
[HLS] Convert the live streaming to HLS for live streaming

### DISCUSSION THREAD

**Discussion by IamBruce828** [pedroSG94/RootEncoder#1679] [Ideas]
Hi Pedro,
I will try to do live streaming from rtmp live streming to hls live streming.
Therefor I will convert the audio/video data to HLS files that updated .m3u8 file and keep generating .ts files. Then upload by https.
I plan to use ffmpeg and do you have any suggestion and ides about HLS conversion in rtmp section ? 
Thanks,
Bruce

**Comment by pedroSG94** [comment]
Hello,

This conversion normally is developed in server side. Most of server allow you reproduce the stream in HLS when you push it in RTMP. For example you can do it with MediaMTX or SRS by default.

**Reply by IamBruce828** [reply]
Because YouTube has a HLS url, I write a python that capture video data from camera, convert the data to HLS files and upload files to the url. It seems work fine.  I think I also can do the same thing by RootEncoder that press the record button and generate hls files.
Is it possible that I  can do a modification for write(int track, ByteBuffer byteBuffer, MediaCodec.BufferInfo info) or anywhere?

**Comment by pedroSG94** [comment]
Hello,

Since you want implement a HLS stream using Youtube you can try create your own StreamBase.
You can create a class similar to this:
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/rtmp/RtmpStream.kt

As you can see that class is using RtmpClient to stream. You can remove RtmpClient and create your own implementation.
The audio/video are received here:
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/rtmp/RtmpStream.kt#L87
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/rtmp/RtmpStream.kt#L91

Let me know if you have any question

**Reply by IamBruce828** [reply]
I will try that. Thanks for your reply.
